pkg/operator/controller/garden/garden/reconciler.go

```diff
@@ -229,10 +229,43 @@ func (r *Reconciler) updateStatusOperationStart(ctx context.Context, garden *ope
 	return nil
 }
 
-func (r *Reconciler) updateStatusOperationSuccess(ctx context.Context, garden *operatorv1alpha1.Garden, conditionReconciled gardencorev1beta1.Condition) error {
-	garden.Status.Conditions = v1beta1helper.MergeConditions(garden.Status.Conditions, v1beta1helper.UpdatedConditionWithClock(r.Clock, conditionReconciled, gardencorev1beta1.ConditionTrue, conditionReasonPrefix(garden)+"Successful", "Garden operation was completed successfully."))
+func (r *Reconciler) updateStatusOperationSuccess(ctx context.Context, garden *operatorv1alpha1.Garden, operationType gardencorev1beta1.LastOperationType) error {
+	var (
+		now                        = metav1.NewTime(r.Clock.Now().UTC())
+		description                string
+		setConditionsToProgressing bool
+	)
+
+	switch operationType {
+	case gardencorev1beta1.LastOperationTypeReconcile:
+		description = "Garden cluster has been successfully reconciled."
+		setConditionsToProgressing = true
+	case gardencorev1beta1.LastOperationTypeDelete:
+		description = "Garden cluster has been successfully deleted."
+		setConditionsToProgressing = false
+	}
+
+	if setConditionsToProgressing {
+		for i, cond := range garden.Status.Conditions {
+			switch cond.Type {
+			case operatorv1alpha1.RuntimeComponentsHealthy,
+				operatorv1alpha1.VirtualComponentsHealthy,
+				operatorv1alpha1.VirtualGardenAPIServerAvailable:
+				if cond.Status != gardencorev1beta1.ConditionFalse {
+					garden.Status.Conditions[i].Status = gardencorev1beta1.ConditionProgressing
+					garden.Status.Conditions[i].LastUpdateTime = metav1.Now()
+				}
+			}
+		}
+	}
```

Comment:
Do we really need this part? We now still have `garden-garden` and `garden-care` reconcilers updating the garden's condition. 
The `cond.Status` in line 254 might be a couple of minutes old when the reconciliation is finished and not up-to-date anymore. Thus, in the end we might overwrite a newer condition based on outdated information.
`garden-care` controller starts reconciling [after a successful reconciliation](https://github.com/gardener/gardener/blob/b5ee7759e60350ae5a245153a061917b5c43f84e/pkg/operator/controller/garden/care/add.go#L121) anyway. Thus, we could avoid conditions jumping back and forth by relying on that, couldn't we?